optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -554,4 +561,171 @@ private static Runnable innerCountDistinctLong(CountDistinctLongResultContainer
     private ConstraintCollectors() {
     }
 
+    // ************************************************************************
+    // toCollection
+    // ************************************************************************
+
+    /**
+     * Creates constraint collector that will return matched facts as a collection.
+     *
+     * @param collectionFunction never null, provides new instance of the target collection with given initial capacity
+     * @param <A> type of the matched fact, matching the type of elements in the resulting collection
+     * @param <Result> type of the target collection
+     * @return never null
+     */
+    public static <A, Result extends Collection<A>> UniConstraintCollector<A, ?, Result> toCollection(
```

Comment:
Do we need toCollection? When in doubt leave it out. I 'd rather start with just toSet() and toList() until someone complains of the lack of toCollection(). Wdyt?

Reply:
I consider it useful for when you want a stable iteration order, so you request `toCollection(TreeSet::new)`.

Reply:
Agreed, or to pick your collection type. Keep it indeed.

This is the special case though. The toList() is the normal case. So toList() gets a full proper javadoc (they others can use "As defined by"). Wdyt?

Reply:
So, we want `toCollection(constructor)`, but not `toCollection()`?